core/src/test/java/org/elasticsearch/action/admin/indices/create/CreateIndexIT.java

```diff
@@ -277,4 +278,19 @@ public void testRestartIndexCreationAfterFullClusterRestart() throws Exception {
         internalCluster().fullRestart();
         ensureGreen("test");
     }
+
+    public void testNoTypeOrTokenizerErrorMessage() throws Exception {
```

Comment:
I think we can test this without creating an index - maybe a simple unittest in `AnalysisServiceTests` would do? To trigger the problem I think all you need to do is to build an `AnalysisService` like this:

```Java
        Version version = VersionUtils.randomVersion(random());
        Settings settings = Settings
            .builder()
            .put(IndexMetaData.SETTING_VERSION_CREATED, version)
            .put(Environment.PATH_HOME_SETTING.getKey(), createTempDir().toString())
            .putArray("analysis.analyzer.test_analyzer.filter", new String[] {"lowercase", "stop", "shingle"});
            .putArray("analysis.analyzer.test_analyzer.char_filter", new String[] {"html_strip"})
            .build();
        IndexSettings idxSettings = IndexSettingsModule.newIndexSettings("index", settings);
        AnalysisService analysisService = new AnalysisRegistry(null, new Environment(settings)).build(idxSettings);
```